docs/assignment2.md

```diff
@@ -2,94 +2,359 @@
+and how classes interact with each other difficult to understand. Some of the relationships have changed since then -- for example, Game and Command don't
+have a composition relationship anymore, but rather a binary association, as a game state has commands, and commands change the game state, so they both reliant
+on each other equally. This made more sense for the structure of the game and also made more logical sense.
+
+You can find the first version of our class diagram below:
+
+![](images/ClassDiagramV1.png)
+
+The **Player** class contains all of the information about the user, and therefore, the player of the game. It contains a *playerStats* object, which sets
+the health points, the name and the inventory of the player. The constructor sets all these values from the information takes from the **PlayerStats** class
+with the approperiate getters and setters for each of the attributes. It also has a *onCommand(Command, Callback)* function which handles all the commands
+that are related to health and inventory. The *listCommands()* function lists all of the  possible commands that the user can type to provide information about
+their progress, such as "stats" and "inventory". The **Player** and **PlayerStats** classes have a composition relationship, with **Player** being composed and contained by
+the **PlayerStats** class. As we can see in the class diagram, **Player** takes on the association with the multiplicity value of 1, and it is the same vice versa.
+This means that a player can only contain one object of *playerStats*, and vice versa, as a player cannot have multiple health points values or multiple inventories. \
+The **PlayerStats** class describes the stats that the player currently has, as well as setting them up in the beginning. The class has the following attributes:
+*minHealthPoints*, *maxHealthPoints*, and *healthPoints*. Respectively, these attributes describe the minimum amount of health points the user can have in
+the game before dying, the maximum amount of health points he can have at any point in the game, and the amount of health points that the user starts out
+with. There are getters and setters for getting this information. There is also a *toString()* method that is overriden from the origin Java function so that
+the stats can be printed on the console should the user type the correct command. The **PlayerStats** inherits from the **Stats** cass. \
+The **Stats** class has a protected attribute *healthPoints*, the getter ans setter methods for that attribute, as well as printing out the stats using
+the *toString()* method. The reason for the inheritance is the possibility of NPCs in the game sharing certain stats with the player, like health points.
+When these NPCs are added, those classes can inherit from the **Stats** class to encourage code re-usability.
+
```

Comment:
-Description: Great
-Model: Ok the diagram projects your description and matches your implementation on a logical level but the modeling elements in some cases are contrived. Some notes:
1)Naming conventions slightly off in some cases i.e Actions is plural but multiplicity is 1 and used in a composition 
2)You can't have compositions when the contained object is actually shared among other classes
3)PlayerStats has a health points attribute and is also inheriting health points from the vanilla Stats class? 
3) Good use of encapsulation and information hiding

Reply:
@sophiak2000 Reminder to look at this.